### PR TITLE
Made suggested updates to functions docs

### PR DESCRIPTION
**Pull request by ekansa**
### brief description of changes
Updates to docs about creating custom functions

#### issues addressed
(https://github.com/archesproject/arches-docs/issues/441)

#### further comments
I haven't actually designed any functions, so I'll need some guidance to make sure this is correct.

To see the updates:

1. Function hooks: https://arches.readthedocs.io/en/funcs_7_5/developing/extending/extensions/functions/#function-hooks
2. Function descriptions:  https://arches.readthedocs.io/en/funcs_7_5/developing/extending/extensions/functions/#defining-the-function-s-details

---

This box **must** be checked
- [x] the PR branch was originally made from the base branch

This box **should** be checked
- [x] after these changes the docs build locally without error

This box **should only** be checked you intend to follow through on it (we can do it on our end too)
- [x] I will `cherry-pick` all commits in this PR into other branches that should have them _after_ this PR is merged

### CODE REVIEW

**Comment by ekansa** [comment]
@bferguso Just a quick ping to see if you could review this PR based on the issue you raise. Thanks!

File: docs/developing/extending/extensions/functions.rst [hunk 1]
```diff
@@ -11,6 +11,17 @@ Functions.
 Function must be created, registered, and then associated with a
 Resource Model.
 
+Functions are similar to database triggers. On get, save, post_save, 
+and delete operations of a tile, the Python code in a function is run. 
+For example, the primary descriptor function saves the primary descriptors 
+of a resource model on the save event of a tile. Hypothetically, you could 
+also do something like update an external system on the creation of a 
+tile or send an email notification.
+
```
Comment (bferguso) [inline docs/developing/extending/extensions/functions.rst:20]: Maybe add a little blurb about the purpose of the primary descriptors function if that isn't called out somewhere else?
The primary descriptors function is used to generate the name of the resource, which is used to identify the resource in places such as the search results card, map popup and report title. It is also used to generate the resource descriptions displayed in the body of the search results and map popup cards.

The resource descriptors are generated on resource instance save. They can also be regenerated for all the resources of a particular type by running the elasticsearch reindex management command with the `--recalculate-descriptors` flag. A description of the elasticsearch management commands can be found here (not sure how the reference is done): https://github.com/archesproject/arches-docs/blob/master/docs/developing/reference/command-line-reference.rst#elasticsearch-management

**Review by bferguso — CHANGES_REQUESTED** [review]
Looks great! It's great to see the types of possible functions called out as it was never fully clear before. 

I wonder if it worth pointing people at the base class files in GitHub so that they stay current? (Not sure what the disposition is between copying code from those into docs vs referencing them directly).

Whichever method is deemed best, maybe also worth referencing the primary descriptors interface, which can be found here: https://github.com/archesproject/arches/blob/c6b043ae0357892f25f1c143daadde1162ff4e19/arches/app/functions/primary_descriptors.py#L12C7-L12C41

**Comment by mradamcox** [comment]
@bferguso @ekansa FWIW, there is an explanation of how to configure the resource descriptors function in the "configuring" section: https://arches.readthedocs.io/en/stable/configuring/define-resource-descriptors/. May be worth linking to from within this new content, or updating.

File: docs/developing/extending/extensions/functions.rst (hunk 1 above)
Comment (ekansa) [reply docs/developing/extending/extensions/functions.rst:20]: @bferguso Thanks! I added that!

**Comment by ekansa** [comment]
@mradamcox Thanks !!! I added a cross reference as you suggested.

File: docs/developing/extending/extensions/functions.rst (hunk 1 above)
Comment (bferguso) [reply docs/developing/extending/extensions/functions.rst:20]: Looks good! Do we want to point people at the base Display Descriptor function API as well? The section that @mradamcox referred to describes how to use the built-in default primary descriptors function, but not how to extend or add a custom one. I feel like if we're going to call out the tile trigger function API below we may want to at least reference the primary descriptors one too?

**Pull request merged**